src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,231 @@
+    *ppSignatureBuilder = pSignatureBuilder.Detach();
+    return S_OK;
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetInstrumentationMethod(_In_ REFGUID cslid, _Out_ IUnknown** ppUnknown)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IUnknown> pUnknown;
+    IfFailRet(m_pProfilerManager->GetInstrumentationMethod(cslid, &pUnknown));
+    *ppUnknown = pUnknown.Detach();
+    return S_OK;
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IInstrumentationMethod> pInstrumentationMethod;
+    IfFailRet(m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        &pInstrumentationMethod));
+    *ppInstrumentationMethod = pInstrumentationMethod.Detach();
+    return S_OK;
+}
+
+// IProfilerManager2 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    HRESULT hr = S_OK;
+    DWORD apiVer;
+    IfFailRet(m_pProfilerManager->GetApiVersion(&apiVer));
+    *pApiVer = apiVer;
+    return S_OK;
+}
+
+// IProfilerManager4 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pLogging;
+    IfFailRet(m_pProfilerManager->GetGlobalLoggingInstance(&pLogging));
+    *ppLogging = pLogging.Detach();
+    return S_OK;
+}
+
+
+// IProfilerManager5 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    HRESULT hr = S_OK;
+    BOOL fRegistered;
+    IfFailRet(m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, &fRegistered));
+    *pfRegistered = fRegistered;
+    return S_OK;
+}
+
+// IProfilerManagerLogging Methods
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
```

Comment:
>LogMessage [](start = 65, length = 10)

Pass format string and guid instead.

Reply:
DIscussed this offline, for internal interface, it's fairly trivial.

The harder task would be if we were to support this interface publicly and how to handle dll boundaries, unnecessary memory allocations, etc. I'll create a task/issue tracking the latter discussion.

---
In reply to: [331712326](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331712326) [](ancestors = 331712326)